lib/webmock/http_lib_adapters/net_http.rb

```diff
@@ -273,8 +273,20 @@ def self.request_signature_from_request(net_http, request, body = nil)
 
       headers = Hash[*request.to_hash.map {|k,v| [k, v]}.inject([]) {|r,x| r + x}]
 
-      headers.reject! {|k,v| k =~ /[Aa]uthorization/ && v.first =~ /^Basic / } #we added it to url userinfo
+      # If you try and make a request with headers that are symbols,
+      # Net Http raises the following NoMethodError.
+      #
+      # WebMock normalizes headers when creating a RequestSignature,
+      # and will update all headers from symbols to strings.
+      # This means that you could have a test passing with WebMock, but failing in reality.
+      header_as_symbol = headers.keys.find {|header| header.is_a? Symbol}
+      if header_as_symbol
+        raise NoMethodError.new(
```

Comment:
The current error message does emulate Net::HTTP actual behavior but is a really confusing error message to get.  I definitely feel @cjlarose 's point about raising an error message that is more descriptive.  I could go either way though :+1: